### PR TITLE
Show users email in profile (#4003)

### PR DESCRIPTION
**Pull request by luke-**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [X] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [X] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] It's submitted to the `develop` branch, _not_ the `master` branch if no hotfix
- [ ] When resolving a specific issue, it's referenced in the PR's description (e.g. `Fix #xxx[,#xxx]`, where "xxx" is the Github issue number)
- [ ] All tests are passing
- [ ] New/updated tests are included
- [ ] Changelog was modified

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Comment by buddh4** [comment]
The virtual email field is only available for new installation isn't it? Do we need a migration for existing installations?

File: protected/humhub/modules/user/models/fieldtype/UserEmail.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+     * @inheritDoc
+     */
+    public function getUserValue($user, $raw = true)
+    {
+        if (empty($user->email)) {
+            return '';
+        }
+
+        if ($raw) {
+            return Html::encode($user->email);
```
Comment (buddh4) [inline protected/humhub/modules/user/models/fieldtype/UserEmail.php:32]: Shouldn't raw return an unencoded value?

File: protected/humhub/modules/user/models/fieldtype/BaseTypeVirtual.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+     */
+    public function rules()
+    {
+        return [];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getUserValue($user, $raw = true)
```
Comment (buddh4) [inline protected/humhub/modules/user/models/fieldtype/BaseTypeVirtual.php:38]: Why not abstract?

File: protected/humhub/modules/user/models/fieldtype/BaseTypeVirtual.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    /**
+     * @inheritdoc
+     */
+    public $isVirtual = true;
+
+    /**
+     * @inheritdoc
+     */
+    public function rules()
```
Comment (buddh4) [inline protected/humhub/modules/user/models/fieldtype/BaseTypeVirtual.php:30]: Is the same as parent `yii\base\Model`

**Comment by buddh4** [comment]
When creating a virtual field you can still select **editable**, **show at registration** etc. even though those settings are not allowed and do not show up when editing such a field.

File: protected/humhub/modules/user/models/fieldtype/BaseTypeVirtual.php (hunk 2 above)
Comment (buddh4) [reply protected/humhub/modules/user/models/fieldtype/BaseTypeVirtual.php:38]: Ok, unfortunately the parent function is not abstract. Maybe we could use a template method to force subclasses to implement a value getter e.g. `abstract getVirtualUserValue()`

**Comment by luke-** [comment]
> When creating a virtual field you can still select **editable**, **show at registration** etc. even though those settings are not allowed and do not show up when editing such a field.
> 
> +See other comments

Unfortunately, this cannot be avoided during creation, but will otherwise be hidden.

File: protected/humhub/modules/user/models/fieldtype/BaseTypeVirtual.php (hunk 2 above)
Comment (luke-) [reply protected/humhub/modules/user/models/fieldtype/BaseTypeVirtual.php:38]: Good idea!

File: protected/humhub/modules/user/models/fieldtype/UserEmail.php (hunk 1 above)
Comment (luke-) [reply protected/humhub/modules/user/models/fieldtype/UserEmail.php:32]: It's always encoded but without any extras like Link tags.
https://github.com/humhub/humhub/blob/master/protected/humhub/modules/user/models/fieldtype/Text.php#L203

**Pull request merged**